compiler/backend_inkwell/src/lib.rs

```diff
@@ -0,0 +1,1041 @@
+use candy_frontend::mir::{Body, Id, Mir};
+use inkwell::{
+    builder::Builder,
+    context::Context,
+    module::{Linkage, Module},
+    support::LLVMString,
+    types::{BasicType, StructType},
+    values::{ArrayValue, BasicValue, BasicValueEnum, FunctionValue, GlobalValue, PointerValue},
+    AddressSpace,
+};
+
+pub use inkwell;
+use std::{
+    collections::{HashMap, HashSet},
+    path::Path,
+    rc::Rc,
+    sync::Arc,
+};
+
+#[derive(Clone)]
+struct FunctionInfo<'ctx> {
+    function_value: Rc<FunctionValue<'ctx>>,
+    captured_ids: Vec<Id>,
+    env_type: Option<StructType<'ctx>>,
+}
+
+pub struct CodeGen<'ctx> {
+    context: &'ctx Context,
+    module: Module<'ctx>,
+    builder: Builder<'ctx>,
+    mir: Arc<Mir>,
+    tags: HashMap<String, Option<Id>>,
+    values: HashMap<Id, Rc<GlobalValue<'ctx>>>,
+    locals: HashMap<Id, Rc<BasicValueEnum<'ctx>>>,
+    functions: HashMap<Id, FunctionInfo<'ctx>>,
+    unrepresented_ids: HashSet<Id>,
+    main_return: Option<PointerValue<'ctx>>,
+}
+
+impl<'ctx> CodeGen<'ctx> {
+    pub fn new(
+        context: &'ctx Context,
+        module: Module<'ctx>,
+        builder: Builder<'ctx>,
+        mir: Arc<Mir>,
+    ) -> Self {
+        Self {
+            context,
+            module,
+            builder,
+            mir,
+            tags: HashMap::new(),
+            values: HashMap::new(),
+            locals: HashMap::new(),
+            functions: HashMap::new(),
+            unrepresented_ids: HashSet::new(),
+            main_return: None,
+        }
+    }
+
+    pub fn compile(
+        mut self,
+        path: &Path,
+        print_llvm_ir: bool,
+        print_main_output: bool,
+    ) -> Result<(), LLVMString> {
+        let i32_type = self.context.i32_type();
+        let i64_type = self.context.i64_type();
+        let i8_type = self.context.i8_type();
+        let void_type = self.context.void_type();
```

Comment:
Nit: I think this order is more intuitive

```suggestion
        let void_type = self.context.void_type();
        let i8_type = self.context.i8_type();
        let i32_type = self.context.i32_type();
        let i64_type = self.context.i64_type();
```